genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/TitusAgentLauncher.java

```diff
@@ -0,0 +1,226 @@
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Agent launcher that spawns a job in a dedicated container through Titus (https://netflix.github.io/titus/).
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+public class TitusAgentLauncher implements AgentLauncher {
+
+    private static final String LAUNCH_TIMER = "genie.launchers.titus.launch.timer";
+    private static final String GENIE_USER_ATTR = "genie_user";
+    private static final String GENIE_SOURCE_HOST_ATTR = "genie_source_host";
+    private static final String GENIE_ENDPOINT_ATTR = "genie_endpoint";
+    private static final String GENIE_JOB_ID_ATTR = "genie_job_id";
+
+    private final RestTemplate restTemplate;
+    private final Cache<String, String> healthIndicatorCache;
+    private final GenieHostInfo genieHostInfo;
+    private final TitusAgentLauncherProperties titusAgentLauncherProperties;
+    private final MeterRegistry registry;
+
+    /**
+     * Constructor.
+     *
+     * @param restTemplate                 the rest template
+     * @param healthIndicatorCache         a cache to store metadata about recently launched jobs
+     * @param genieHostInfo                the metadata about the local server and host
+     * @param titusAgentLauncherProperties the configuration properties
+     * @param registry                     the metric registry
+     */
+    public TitusAgentLauncher(
+        final RestTemplate restTemplate,
+        final Cache<String, String> healthIndicatorCache,
+        final GenieHostInfo genieHostInfo,
+        final TitusAgentLauncherProperties titusAgentLauncherProperties,
+        final MeterRegistry registry
+    ) {
+        this.restTemplate = restTemplate;
+        this.healthIndicatorCache = healthIndicatorCache;
+        this.genieHostInfo = genieHostInfo;
+        this.titusAgentLauncherProperties = titusAgentLauncherProperties;
+        this.registry = registry;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<JsonNode> launchAgent(
+        final ResolvedJob resolvedJob,
+        @Nullable final JsonNode requestedLauncherExt
+    ) throws AgentLaunchException {
+
+        final String jobId = resolvedJob.getJobSpecification().getJob().getId();
+
+        final TitusBatchJobRequest titusJobRequest = this.createJobRequest(resolvedJob);
+        final Set<Tag> tags = Sets.newHashSet();
+
+        String titusJobId = "unknown";
+
+        final long start = System.nanoTime();
+        try {
+            final TitusBatchJobResponse titusResponse = this.restTemplate.postForObject(
+                this.titusAgentLauncherProperties.getEndpoint(),
```

Comment:
This could be a constant assigned in the constructor since it's not allowed to change right?